.github/workflows/coveralls.yaml

```diff
@@ -0,0 +1,31 @@
+name: Coveralls report
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    paths:
+      - 'subber/**'
+      - 'adder/**'
+      - 'go.mod'
+
+jobs:
+  report:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Wait on tests
+        uses: lewagon/wait-on-check-action@v1.3.1
+        with:
+          ref: ${{ github.sha }}
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+          wait-interval: 10 # seconds
+          running-workflow-name: report
+          allowed-conclusions: success,skipped,cancelled,failure
+
+      - uses: coverallsapp/github-action@master
+        env:
+          COVERALLS_SERVICE_NUMBER: ${{ github.sha }}
+          COVERALLS_SERVICE_JOB_ID: ${{ github.sha }}
```

Comment:
In future releases of coveralls github action `COVERALLS_SERVICE_NUMBER` will be used. Now `COVERALLS_SERVICE_JOB_ID` is used. It makes sense to provide both (or switch to `coverallsapp/github-action@v1` and use only `COVERALLS_SERVICE_JOB_ID`)